packages/base/src/mainview/mainView.tsx

```diff
@@ -1661,7 +1661,7 @@ export class MainView extends React.Component<IProps, IStates> {
       }
 
       if (data && (val.action === 'add' || val.action === 'update')) {
-        const jsonData = JSON.parse(data);
+        const jsonData = data as IAnnotation;
```

Comment:
I believe this change will break project files saved with older versions of JGIS. Related: #220

Reply:
I think it makes sense for now to update our project files and mark this as a breaking change in the changelog (perhaps in bold, with some :warning: or :rotating_light: indicator) and just update our example project files.

But I think it would be nice to have a documented policy for breaking changes so users know what to expect and so we can eventually evolve that policy to be more user-friendly!

Reply:
We can also automatically parse the data if it's a string.

```typescript
if (typeof data === 'string') {
  jsonData = JSON.parse(data);
} else {
  jsonData = data;
}
```

Reply:
I agree this is probably the best path forward; it's simple and avoids breaking change.

But we should leave a comment in the code to remind us why we're doing that, and link back to this PR in the comment. After a certain amount of time, we can feel comfortable about JGIS' ability to detect incompatible schemas and feel OK that we don't support schemas that old, and remove this code.

Reply:
Yes, had the same thing in mind, thank you and Martin for suggesting.